migrations/emailmonitor/oxa_email_notifier.py

```diff
@@ -0,0 +1,239 @@
+        fl = str(row[6])
+        rs = str(row[7])
+        f.write('{:6s} {:15s} {:15s} {:15s} {:25s} {:15s}'.format(d,n,a,na,fl,rs)+"\r\n")
+        if ( fl > 0 ):
+            cursor.execute("SELECT * FROM oxa.oxa_activationfailed WHERE xyear="+y+" and xmonth="+m+" and xday="+d)
+            results = cursor.fetchall()
+            for row in  results:
+                resent = ""
+                if row[9] == 1:
+                    resent = " | Resent on " + row[10].strftime("%Y-%m-%d %H:%M:%S")
+                f.write("  "+row[5]+" | " + row[6] + " | "+row[7].strftime("%Y-%m-%d %H:%M:%S") + " | " + row[8].strftime("%Y-%m-%d %H:%M:%S") + " | " +row[11]+resent+"\r\n");
+               
+    f.write("\r\n\r\n")
+     
+    cursor.close()
+    db.close()
+
+def OutputAllFailedEmailActivation( f ):
+    db = MySQLdb.connect(IP,User,Password,Database)
+    cursor = db.cursor()
+
+    dt = datetime.now()
+
+    sql = "SELECT * FROM oxa.oxa_activationfailed \
+          WHERE is_processed=0 \
+          ORDER BY xyear,xmonth,xday,hostvmip"
+
+    cursor.execute(sql)
+    results = cursor.fetchall()
+    f.write("OUTSTANDING FAILED ACTIVATION EMAILS:\r\n\r\n")
+    f.write("=============================\r\n")
+    f.write('{:8s} {:6s} {:6s} {:50s} {:36s} {:24s} {:24s} {:20s}'.format("YEAR","MONTH","DAY","EMAIL","ACTIVATION KEY","DATE JOINED","DATE FAILED","VM IP")+"\r\n")
+    for row in results:
+        y = str(row[1])
+        m = str(row[2])
+        d = str(row[3])
+        line = '{:8s} {:6s} {:6s} {:50s} {:36s} {:24s} {:24s} {:20s}'.format(y,m,d,row[5],row[6],row[7].strftime("%Y-%m-%d %H:%M:%S"),row[8].strftime("%Y-%m-%d %H:%M:%S"),row[11])
+        f.write(line +"\r\n");
+
+    f.write("\r\n\r\n")
+    cursor.close()
+    db.close()
+
+def GetNotProcessedEmailActivationFailures():
+    db = MySQLdb.connect(IP,User,Password,Database)
+    cursor = db.cursor()
+
+
+    sql = "SELECT * FROM oxa.oxa_activationfailed \
+          WHERE user_id > 0 and is_processed=0 \
+          ORDER BY xyear,xmonth,xday,hostvmip"
+
+    cursor.execute(sql)
+    results = cursor.fetchall()
+    cursor.close()
+    db.close()
+
+    return results
+
+def GetMailServerConnection():
+    mailserver = SMTP("smtp-server",port)
+    mailserver.starttls()
+    mailserver.login("UserName", "Password")
+    return mailserver
+
+def SendMail(mailserver, message, toMail, subject, attachment=None):
+    from_addr = ""
+    msg = MIMEMultipart()
+    msg['From'] = ""
+    msg['To'] = toMail
+    msg['Subject'] = subject
+    msg.attach(MIMEText(message))
+
+    if ( attachment != None ):
+        for filename in attachment:  
+            with open("/oxa/"+filename,'r') as f:
+                data = f.read()
+                cnt = MIMEText(data)
+                cnt.add_header('Content-Disposition', 'attachment', filename=filename)
+                msg.attach(cnt)
+
+    mailserver.sendmail(from_addr,toMail,msg.as_string())
+
+def WriteLog( log, error=None ):
```

Comment:
Stephen this is a temp and short-term solution. I used simple approach. I don't think we need to invest more on that. I think it is better with deterministic approach that we know the log file location and name.